eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -444,10 +444,16 @@ object MultiPartHandler {
     val minExpiry = nodeParams.channelConf.minFinalExpiryDelta.toCltvExpiry(nodeParams.currentBlockHeight)
     if (add.cltvExpiry < minExpiry) {
       log.warning("received payment with expiry too small for amount={} totalAmount={}", add.amountMsat, payload.totalAmount)
-      false
-    } else {
-      true
+      return false
+    }
+    if (payload.isInstanceOf[FinalPayload.Blinded]) {
+      val expectedExpiry = payload.expiry + nodeParams.channelConf.minFinalExpiryDelta
```

Comment:
This is unreliable: if a node operator increases their `minFinalExpiryDelta` in their `eclair.conf`, they will reject all payments for invoices created before the change. It also wouldn't work if at some point we let the user use a different value than the one from node params in their blinded path creation. We would need to compare this to the final expiry delta used for this specific invoice, not the one from node params, but we don't store this information for Bolt 12 invoices.

Anyway I don't think this check is very important: since we check in `PaymentPacket.scala` that `add.cltvExpiry < payload.expiry`, intermediate nodes can decrease the cltv expiry by at most `min_final_expiry_delta`, which should be much smaller than any real hop's `cltv_expiry_delta`, so shouldn't let intermediate nodes probe their position.

I've removed this in [TODO](https://github.com/ACINQ/eclair/commit/26b427ad7f2bea6e365851eacf32a3e044b43c4a).

I'm noticing just now that this issue already exists a few lines above for Bolt 11 invoices: we should use the invoice's `minFinalExpiryDelta` instead of the one from `nodeParams`. This should be fixed in a separate PR.

Reply:
Given that invoices expire relatively quickly (and Bolt 12 invoices can expire even quicker) and the `minFinalExpiryDelta` should be very rarely updated, I think it's fine.
Actually, if I ever increase my `minFinalExpiryDelta`, it means I don't feel safe with the old one. So I wouldn't want to continue using the old one, even for already created invoices.

Reply:
Right, that may be surprising behavior for node operators, but I agree with you that it makes sense.

We should still compare to the current block height though, I fixed that in https://github.com/ACINQ/eclair/commit/65624656e6390b03988d983f37e9529987562534 and re-added the payment packet test.

Reply:
Added your commit.

> We should still compare to the current block height though

I was already doing it.

Reply:
True, but the code was more clumsy, it's cleaner now.